include/test/mir/test/geometry_matchers.h

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright © 2022 Canonical Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 or 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef MIR_TEST_GEOMETRY_MATCHERS_H_
+#define MIR_TEST_GEOMETRY_MATCHERS_H_
+
+#include <gmock/gmock.h>
+
+namespace mir
+{
+namespace test
+{
+
+MATCHER_P(TopLeftEq, expected_top_left, "")
+{
+    return arg.top_left == expected_top_left;
+}
+
+MATCHER_P(SizeEq, expected_size, "")
```

Comment:
These matcher names are probably a bit too generic for something in a public header: They require `arg` to be a `Rectangle` (or similar) but there's no mention of that constraint in the naming (or scope)